lab/src/raft/request_vote.go

```diff
@@ -0,0 +1,86 @@
+package raft
+
+//
+// example RequestVote RPC arguments structure.
+// field names must start with capital letters!
+//
+type RequestVoteArgs struct {
+	// Your data here (2A, 2B).
+	Term int
+	CandidateId int
+	LastLogIndex int
+	LastLogTerm int
+}
+
+//
+// example RequestVote RPC reply structure.
+// field names must start with capital letters!
+//
+type RequestVoteReply struct {
+	// Your data here (2A).
+	Term int
+	VoteGranted bool
+}
+
+//
+// example RequestVote RPC handler.
+//
+func (rf *Raft) RequestVote(args *RequestVoteArgs, reply *RequestVoteReply) {
+	// Your code here (2A, 2B).
+	rf.mu.Lock()
+
+	localLastLogIndex := len(rf.log)
+	localLastLogTerm := 0
+	if localLastLogTerm > 0 {
```

Comment:
Should this be `localLastLogIndex` instead? Since I think you are trying to get the term of the last log when it exists by checking if `len(rf.log) > 0`?

`localLastLogTerm` has been set to 0 on line #33 and it will never be larger than 0 here.

Reply:
Ur right, it is a typo